pkg/ccl/streamingccl/streamingest/alter_replication_job.go

```diff
@@ -399,6 +399,48 @@ func alterTenantRestartReplication(
 	), "creating replication job")
 }
 
+func pickReplicationResume(
```

Comment:
i'm finding this function very hard to follow. could you split this into two functions: 1 for resuming a replication stream on the destination tenant after it has cutover; the other for reversing the direction of replication? Once we have the two different functions, I think it will be easier to reason about when we need to throw an error to the user.

Reply:
These are not actually two distinct cases though. They can both occur at once in which case you want to pick the better one, or neither could be true, in which case you have an error.

Imagine:
T0: A replicating from B.
T1: B runs cuts over.
T2: A starts replicating from B (revert to T1 and resume, aka reversal)
T3: A cuts over.
T4: A resumes replicating from B (from either T1 or T3).

In this case it is both a reversal of a prior replication -- new src's replicatedFrom is the new dest's srcID -- which ended at T1 _and_ a resumption of replication of the same src from which it was previously replicating (until T3), so it can resume from either T3 or T1. T3 is later, so it should use that.

Reply:
I see, and we could have a timeline in which the reversal occurs  after the destination cutover; i.e:

T0: A replicating from B.
T1: B runs cuts over.
T2: A starts replicating from B (revert to T1 and resume, aka reversal)
T3: A cuts over.
T4 : B resumes replicating from T3, instead of T1

 Perhaps for the dull, you could add a docstring to this function describing these two timelines. Further, I think the function could be easier to read if we restructured it to compute `latestCutoverOnCurrentCluster` and `latestCutoverOnSourceCluster`, then return the latest of the two or error if both are empty.

Reply:
I had some comments inline but I've adjusted them to use the terms reversal and resume explicitly and am keeping the timestamps separate until the end for clarity. WDYT?

Reply:
This looks much clearer. thanks for changing!